dll/win32/shell32/brfolder.cpp

```diff
@@ -804,6 +805,84 @@ BrFolder_OnCommand(BrFolder *info, UINT id)
     }
 }
 
+static void
+GetTreeViewItemContextMenuPos(HWND hWnd, HTREEITEM hItem, POINT *ppt)
+{
+    RECT rc;
+    if (TreeView_GetItemRect(hWnd, hItem, &rc, TRUE))
+    {
+        ppt->x = (rc.left + rc.right) / 2;
+        ppt->y = (rc.top + rc.bottom) / 2;
+    }
+    ClientToScreen(hWnd, ppt);
+}
+
+static void
+BrFolder_OnContextMenu(BrFolder &info, LPARAM lParam)
+{
+    enum { IDC_EXPAND = 1, ID_FIRSTCMD, ID_LASTCMD = 0xffff };
+    HTREEITEM hSelected = TreeView_GetSelection(info.hwndTreeView);
+    CMINVOKECOMMANDINFOEX ici = { sizeof(ici), CMIC_MASK_PTINVOKE, info.hWnd };
+    ici.fMask |= GetKeyState(VK_SHIFT) < 0 ? CMIC_MASK_SHIFT_DOWN : 0;
```

Comment:
For fool safe, add parens:

```suggestion
    ici.fMask |= (GetKeyState(VK_SHIFT) < 0) ? CMIC_MASK_SHIFT_DOWN : 0;
```

Reply:
> For fool safe

For what? `<` has higher precedence than `?:`. Or are you saying the C++ language rules might change and break millions of lines of code?

Reply:
OK.